test/src/discover_tasks_test.dart

```diff
@@ -147,6 +148,40 @@ main() {
     });
   });
 
+  group('execute returns value', () {
+    grinderTest('task from method', () {}, (MockGrinderContext ctx) {
+      var annotated =
+          discoveryGood.discoverDeclaration(goodLib.declarations[#method], {});
+      var task = annotated.task;
+      expect(task.name, 'method');
```

Comment:
As far as I can tell, the tests look good.  They are testing the the correct `taskFunction` is generated based on the declaration.  Except no need to test the names, since that's already tested elsewhere.

We could also add tests that the return value from arbitrary `taskFunction`s are used, and to test that `Grinder.start` properly waits for futures returned from `GrinderTask.execute`.  But no need to do that in this PR.